gio/src/file.rs

```diff
@@ -1111,6 +1113,137 @@ pub trait FileExtManual: IsA<File> + Sized {
 
         (fut, Box::pin(receiver))
     }
+
+    #[doc(alias = "g_file_set_attribute")]
+    fn set_attribute(
+        &self,
+        attribute: &str,
+        type_value: FileAttributeTypeValue,
+        flags: FileQueryInfoFlags,
+        cancellable: Option<&impl IsA<Cancellable>>,
+    ) -> Result<(), glib::Error> {
+        unsafe {
+            let mut error = std::ptr::null_mut();
+            let is_ok = ffi::g_file_set_attribute(
+                self.as_ref().to_glib_none().0,
+                attribute.to_glib_none().0,
+                type_value.type_().into_glib(),
+                type_value.to_glib_none().0,
+                flags.into_glib(),
+                cancellable.map(|p| p.as_ref()).to_glib_none().0,
+                &mut error,
+            );
+            debug_assert_eq!(is_ok == glib::ffi::GFALSE, !error.is_null());
+            if error.is_null() {
+                Ok(())
+            } else {
+                Err(from_glib_full(error))
+            }
+        }
+    }
 }
 
 impl<O: IsA<File>> FileExtManual for O {}
+
+#[derive(Debug)]
+// checker-ignore-item
+pub enum FileAttributeTypeValue<'a> {
+    Pointer(FileAttributeType, glib::ffi::gpointer),
+    String(&'a str),
+    ByteString(&'a str),
```

Comment:
Isn't this supposed to be a `&[u8]`?

Reply:
It's actually a NUL-terminated string of some sort, not a literal byte string where there can be 0-bytes in the middle?